cpp/include/mcts/UniformNNEvaluationService.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <core/concepts/Game.hpp>
+#include <mcts/NNEvaluation.hpp>
+#include <mcts/NNEvaluationRequest.hpp>
+#include <mcts/NNEvaluationServiceBase.hpp>
+
+template <core::concepts::Game Game>
+class UniformNNEvaluationService : public mcts::NNEvaluationServiceBase<Game> {
+ public:
+  using NNEvaluation = mcts::NNEvaluation<Game>;
+  using NNEvaluationRequest = mcts::NNEvaluationRequest<Game>;
+  using ValueTensor = NNEvaluation::ValueTensor;
+  using PolicyTensor = NNEvaluation::PolicyTensor;
+  using ActionValueTensor = NNEvaluation::ActionValueTensor;
+  using ActionMask = NNEvaluation::ActionMask;
+
+  void evaluate(const NNEvaluationRequest& request) override {
+    ValueTensor value;
+    PolicyTensor policy;
+    ActionValueTensor action_values;
+    group::element_t sym = group::kIdentity;
+
+    for (typename NNEvaluationRequest::Item& item : request.items()) {
+      ActionMask valid_actions = item.node()->stable_data().valid_action_mask;
+      core::seat_index_t cp = item.node()->stable_data().current_player;
+
+      int n = valid_actions.count();
+      float p = 1.0 / n;
+
+      policy.setZero();
+      for (int i = 0; i < policy.size(); i++) {
+        if (valid_actions[i]) {
+          policy[i] = p;
+        }
+      }
```

Comment:
You don't actually need this `for` loop at all. Remember, the `NNEValuation` constructor expects *raw* values, and performs a softmax on those raw values. Furthermore, it masks according to `valid_actions`.

So all that matters here is that `policy` is uniform across the set of valid actions, and `policy.setZero()` is sufficient for that.